events/fetcher.go

```diff
@@ -125,13 +126,19 @@ func (m *Fetcher) startStream(app cfclient.App) chan cfclient.App {
 
 		msgs, errs := conn.Stream(app.Guid, authToken)
 
-		m.newAppChan <- app.Guid
+		for i := 0; i < app.Instances; i++ {
+			m.newAppChan <- fmt.Sprintf("%s:%d", app.Guid, i)
```

Comment:
I wonder if instead of sending a string appending the new instance, we should refactor these channels to send the App struct itself (or a new struct with the right info).

So the channel would be a  `chan cfclient.App`

The same for the updatedApp chan. 

I think the `updatedApp` channel should receive messages only if the app attributes we are interested on have changed (e.g. number of instances).

Reply:
I think the fact that the channel should receive messages only if the app attributes we are interested in have changed shows that this would be a leaky abstraction: the fetcher has to know which attributes the sender cares about, so the fetcher still has to have the same conditional logic that it has now.

I think it's probably fine as it is.